### PR TITLE
Make build_lbann.sh script extensible

### PR DESCRIPTION
**Pull request by bvanessen**
Added a new --extras or -e flag to allow users to include a file of
Spack packages that should be included in the environment.  Added an
option to only configure and to print help directions when only
configuring.  Also cleaned up some of the other flags and their help
message.

### CODE REVIEW

**Comment by bvanessen** [comment]
Here is an example command of how to use this `./scripts/build_lbann.sh -d -e scripts/external_packages_atom.sh -- +cuda`

**Review by samadejacobs — APPROVED** [review]
WGFM

**Review by benson31 — APPROVED** [review]
LGTM

File: docs/building_lbann.rst [hunk 1]
```diff
@@ -62,10 +62,52 @@ Setup Spack (One-time setup)
           `Spack Documentation
           <https://spack.readthedocs.io/en/latest/configuration.html>`_.
 
+
+.. _install_lbann_as_user_known_center:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Building & Installing LBANN as a user at a known HPC Center
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (benson31) [inline docs/building_lbann.rst:70]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Building & Installing LBANN as a user at a known HPC Center
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: docs/building_lbann.rst [hunk 2]
```diff
@@ -62,10 +62,52 @@ Setup Spack (One-time setup)
...
+   .. code-block: bash
+
+      spack load lbann@<version>
+
 .. _install_lbann_as_user:
 
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-Building & Installing LBANN as a user
+Building & Installing LBANN as a user (with maximum flexiblity)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (benson31) [inline docs/building_lbann.rst:111]: I can't make a suggestion here for some reason, but the `~` need to go beyond the header text. Add more tildes to lines 109 and 111.

**Review by benson31 — APPROVED** [review]
LGTM

**Pull request merged**